packages/config/src/ProfileInfo.ts

```diff
@@ -524,19 +641,54 @@ export class ProfileInfo {
-                        argValue: (propObj as ICommandProfileProperty).optionDefinition?.defaultValue,
-                        argLoc: { locType: ProfLocType.DEFAULT },
-                        secure: propObj.secure
-                    });
+                    let argFound = false;
+                    if (profile.profLoc.locType === ProfLocType.TEAM_CONFIG) {
+                        let [argLoc, foundInSecureArray]: [IProfLoc, boolean] = [null, false];
+                        try {
+                            [argLoc, foundInSecureArray] = this.argTeamConfigLoc(profile.profName, propName);
+                            argFound = true;
+                        } catch (_argNotFoundInServiceProfile) {
+                            if (this.mLoadedConfig.api.profiles.defaultGet("base")) {
+                                try {
+                                    [argLoc, foundInSecureArray] = this.argTeamConfigLoc(this.mLoadedConfig.properties.defaults.base, propName);
+                                    argFound = true;
+                                } catch (_argNotFoundInBaseProfile) {
+                                    // Do nothing
+                                }
+                            }
+                        }
+                        if (argFound) {
+                            const newArg: IProfArgAttrs = {
+                                argName: propName,
+                                dataType: this.argDataType(propInfoInSchema.type),
+                                argValue: (propInfoInSchema as ICommandProfileProperty).optionDefinition?.defaultValue,
+                                argLoc,
+                                // See https://github.com/zowe/imperative/issues/739
+                                secure: foundInSecureArray || propInfoInSchema.secure
+                            };
+                            try {
+                                this.loadSecureArg({ argLoc, argName: propName } as any);
+                                mergedArgs.knownArgs.push(newArg);
+                            } catch (_secureValueNotFound) {
+                                mergedArgs.missingArgs.push(newArg);
+                            }
+                        }
+                    }
+                    if (!argFound) {
+                        mergedArgs.missingArgs.push({
+                            argName: propName,
+                            dataType: this.argDataType(propInfoInSchema.type),
+                            argValue: (propInfoInSchema as ICommandProfileProperty).optionDefinition?.defaultValue,
+                            argLoc: { locType: ProfLocType.DEFAULT },
+                            secure: propInfoInSchema.secure
+                        });
+                    }
                 } else {
-                    knownArg.secure = propObj.secure;
+                    knownArg.secure = knownArg.secure ?? propInfoInSchema.secure;
```

Comment:
changing this line to `knownArg.secure = knownArg.secure || propInfoInSchema.secure;` will revert:
- #755

Reply:
After some discussion with both CLI and ZE squads...
Since the user went out of their way to make the property insecure, the API should respect that and return schema-defined secure properties (stored in plain-text by the user) when requested with `getSecureVals: false`.